### DISCUSSION TITLE
How to show light mode and dark mode logo?

### DISCUSSION THREAD

**Discussion by iammuttaqi** [saadeghi/daisyui#2493] [Q&A] [answered]
How to toggle between two logo?

for example, I can toggle between logo using `dark` variant:
```html
<img alt="Light Mode Logo" class="block w-48 dark:hidden" src="/images/logo.png">
<img alt="Dark Mode Logo" class="hidden w-48 dark:block" src="/images/logo-white.png">
```

Since DaisyUI's approach is different. How can I achieve that in DaisyUI?

**Comment by saadeghi** [comment, marked as answer]
You can use a class name like this:
```
[[data-theme=dark]_&]:hidden
```
Since daisyUI theme is defined like `<html data-theme="dark">`, `[[data-theme=dark]_&]:hidden` class hides the element if the parent has `data-theme="dark"`.

Example:  
https://play.tailwindcss.com/32M1pqHLPt

**Comment by olli757** [comment]
I tried this in a Next.js project and it doesn't hide it. Does this work in Next.js?
(the html tag doesn't have the data-theme tag when I check in dev tools)

for info, the relevant sections in my Tailwind config file are:

```
daisyui: {
    themes: ["light", "dark"],
  },
  darkMode: ["selector", '[data-theme="dark"]']
```

Thanks!
Olivier

**Reply by saadeghi** [reply]
>Does this work in Next.js? 

Framework doesn't matter.

Can you share a small reproduction link?

**Comment by olli757** [comment]
what would be the best way to reproduce this?

I\`m using DaisyUI out of the box, and it automatically detects if the browser/system has light or dark mode. So I don't set the data-theme myself. (and it doesn't seem to be set at all in dev tool either, hence why ` [[data-theme=dark]_&]:hidden ' doesn't work). 

Question is, how does DaisyUI know how to set light or dark, and how can I display another icon depending on light/dark?

**Reply by saadeghi** [reply]
If you're not using `data-theme` or you're not changing the theme using JS, by default `light` theme is active and if browser/OS is in dark mode, `dark` theme will be active because of `@media (prefers-color-scheme: dark)`

In this case you can use Tailwind's `dark:` prefix which applies styles for `@media (prefers-color-scheme: dark)` only.

For example:
```
<div class="dark:hidden block">only light</div>
<div class="hidden dark:block">only dark</div>
```

**Comment by olli757** [comment]
Ok thanks a lot for this. And also for creating DaisyUI.

Finally I found I had to remove the line `darkMode: ["selector", '[data-theme="dark"]']` from tailwind.config.js, in order to have the dark: selector in tailwind work again.

I never thought of this as the daisyUI docs told me to put it in.

[https://daisyui.com/docs/themes/#-9 ](https://daisyui.com/docs/themes/#-9) How to apply Tailwind's 'dark:' selector for specific themes

Maybe adding one line of explanation/warning about this might help some people
